wandb/sdk/wandb_settings.py

```diff
@@ -1543,6 +1545,7 @@ def _apply_env_vars(
             "WANDB_NOTES": "run_notes",
             "WANDB_TAGS": "run_tags",
             "WANDB_JOB_TYPE": "run_job_type",
+            "WANDB_LAUNCH_JOB_NAME": "job_name",
```

Comment:
Why not call the setting `launch_job_name`? No such mapping would be required then (any env var starting with WANDB_ gets mapped to a setting automatically).